arangod/Agency/State.cpp

```diff
@@ -843,24 +863,40 @@ bool State::loadCollections(TRI_vocbase_t* vocbase,
   return false;
 }
 
+
 /// Load actually persisted collections
 bool State::loadPersisted() {
   TRI_ASSERT(_vocbase != nullptr);
 
   if (!checkCollection("configuration")) {
-    createCollection("configuration");
+    createCollection("configuration", false);
   }
 
   loadOrPersistConfiguration();
 
   if (checkCollection("log") && checkCollection("compact")) {
-    bool lc = loadCompacted();
-    bool lr = loadRemaining();
-    return (lc && lr);
+    index_t lastCompactionIndex = loadCompacted();
+    if (loadRemaining(lastCompactionIndex)) {
+      return true;
+    } else {
+      LOG_TOPIC("1a476", INFO, Logger::AGENCY)
+        << "Non matching compaction and log indexes. Dropping both collections";
+      _log.clear();
+      _cur = 0;
+      dropCollection("log");
+      dropCollection("compact");
+
+    }
   }
 
-  LOG_TOPIC("9e72a", DEBUG, Logger::AGENCY) << "Couldn't find persisted log";
-  createCollections();
+  LOG_TOPIC("9e72a", DEBUG, Logger::AGENCY) << "No persisted log: creating collections.";
+
+  // This is a combined create of logs and compact, as otherwise inconsistencies
+  // in log and compact cannot be mitigated after this point. We are here because
+  // of the above missing / incomplete log / compaction. Including the case of only
+  // one of the two collections being present.
+  createCollection("log", true);
```

Comment:
If we get here because only one of the collections exist, then we are not clearing `_log` nor resetting `_cur` to 0.

Reply:
No, but it is still empty and `_cur == 0` because we are very early in the startup. And when both were there and `loadRemaining` has failed, then we delete `_log` again and set `_cur` to 0.